indigo_api/models/documents.py

```diff
@@ -544,6 +548,33 @@ def change_date(self, new_date, user, comment=None):
             self.expression_date = new_date
             self.save_with_revision(user, comment=comment)
 
+    def update_provision_xml(self, provision_eid, provision_xml):
+        provision = etree.fromstring(provision_xml)
+        # TODO: a better way to discard the wrapping akn tag (using xpath / getchildren()?)
+        actual_provision = list(provision)[0]
+
+        # TODO: don't need this, just for debugging for now
+        actual_provision_str = etree.tostring(actual_provision, encoding='unicode', pretty_print=True)
+
+        main_tree = self.doc.main
+        # TODO: a better way to grab the (only) element with the given provision_eid
+        elems = main_tree.xpath(f".//a:*[@eId='{provision_eid}']", namespaces={'a': self.doc.namespace})
```

Comment:
you can use `self.doc.get_portion_element()` which is a method on `cobalt.StructuredDocument`

Reply:
This is what we use when fetching portions for popups etc.